x/erc20/keeper/precompiles.go

```diff
@@ -4,64 +4,52 @@
 package keeper
 
 import (
+	"fmt"
+	"slices"
+
 	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/vm"
 	"github.com/evmos/evmos/v18/precompiles/erc20"
-	"github.com/evmos/evmos/v18/precompiles/werc20"
 	"github.com/evmos/evmos/v18/x/erc20/types"
 )
 
-// RegisterERC20Extensions registers the ERC20 precompiles with the EVM.
-func (k Keeper) RegisterERC20Extensions(ctx sdk.Context) error {
-	precompiles := make([]vm.PrecompiledContract, 0)
-	params := k.evmKeeper.GetParams(ctx)
-	evmDenom := params.EvmDenom
-
-	var err error
-	k.IterateTokenPairs(ctx, func(tokenPair types.TokenPair) bool {
-		// skip registration if token is native or if it has already been registered
-		// NOTE: this should handle failure during the selfdestruct
-		if tokenPair.ContractOwner != types.OWNER_MODULE ||
-			k.evmKeeper.IsAvailablePrecompile(tokenPair.GetERC20Contract()) {
-			return false
-		}
-
-		var precompile vm.PrecompiledContract
-
-		if tokenPair.Denom == evmDenom {
-			precompile, err = werc20.NewPrecompile(tokenPair, k.bankKeeper, k.authzKeeper, *k.transferKeeper)
-		} else {
-			precompile, err = erc20.NewPrecompile(tokenPair, k.bankKeeper, k.authzKeeper, *k.transferKeeper)
-		}
-
-		if err != nil {
-			err = errorsmod.Wrapf(err, "failed to instantiate ERC-20 precompile for denom %s", tokenPair.Denom)
-			return true
-		}
-
-		address := tokenPair.GetERC20Contract()
+// GetDynamicPrecompileInstance returns the dynamic precompile instance for the given address.
+func (k Keeper) GetERC20PrecompileInstance(
```

Comment:
Nit: add tests for these functions.
We can add these on another PR, but mentioning it so we create a ticket to track this and make sure to include them